feature/learningSuite -> dev

geogor37 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/byu-oit/android-byu-suite-v2/501)
<!-- Reviewable:end -->

romrell4 commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 37 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30fCZ62JsSvNagR-gb:-L30fCZ62JsSvNagR-gc:byp2kum) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L37)):*
> ```Java
> 	public static final String COURSE_NAME = "courseName";
> 
> 	private YearTerm currentYearTerm;
> ```

Is there a way that this could be saved in the spinner? How is the currentYearTerm

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 38 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30fSiGQsqE5SaJ5twN:-L30fSiGQsqE5SaJ5twO:b-n5bwa8) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L38)):*
> ```Java
> 
> 	private YearTerm currentYearTerm;
> 	private HashMap<YearTerm, List<Course>> studentCourses;
> ```

Use Map here instead of HashMap (always use the interface)

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 48 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30gFKsGQKKBttbmb0s:-L30gFKsGQKKBttbmb0t:b-8grjvs) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L48)):*
> ```Java
> 
> 		// We need the current year term to make the initial learning suite calls
> 		ControlDatesClientHelper.getCurrentYearTerm(this, new ByuCallback<ControlDatesResponseWrapper>(this) {
> ```

This should be removed now. Just use the "greatest" year term returned from the service

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 76 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30gUTSDctFEIXhrugm:-L30gUTSDctFEIXhrugn:byp18fd) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L76)):*
> ```Java
> 			enqueueCall(api.getCoursesForInstructor(personId), new CourseListCallback(this, list -> {
> 				instructorCourses = createCourseMap(CollectionUtils.filter(list, course -> course == null
> //						|| !yearTerm.equals(course.getPeriod())
> ```

Remove

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 83 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30h_5L_ZxhBXkg0JFO:-L30h_5L_ZxhBXkg0JFP:buj2gn0) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L83)):*
> ```Java
> 				@Override
> 				public void onSuccess(Call<ArrayList<Course>> call, Response<ArrayList<Course>> response) {
> 					ArrayList<Course> studentCourses = response.body();
> ```

use List

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 106 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30haZvN6aQeUAEOX_K:-L30haZvN6aQeUAEOX_L:bx2a1z2) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L106)):*
> ```Java
> 	}
> 
> 	private HashMap<YearTerm, List<Course>> createCourseMap(List<Course> courses) {
> ```

use Map

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 107 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30hcGSc3ZURrkOt7Lr:-L30hcGSc3ZURrkOt7Ls:b-uphajh) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L107)):*
> ```Java
> 
> 	private HashMap<YearTerm, List<Course>> createCourseMap(List<Course> courses) {
> 		HashMap courseMap = new HashMap<>();
> ```

Map and parameterize <YearTerm, List<Course>>

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 113 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iCkiV4ZTXyawGKaD:-L30iCkiV4ZTXyawGKaE:b-mhqsk3) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L113)):*
> ```Java
> 				courseMap.put(courseYearTerm, new ArrayList<>());
> 			}
> 			((List<Course>) courseMap.get(courseYearTerm)).add(course);
> ```

If you parameterize, you shouldn't have to cast here

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 131 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iQ9HX8cwgBaOj6Zt:-L30iQ9HX8cwgBaOj6Zu:b-kes362) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L131)):*
> ```Java
> 			View view = ((ViewStub) findViewById(R.id.ls_two_fragments_stub)).inflate();
> 			ViewPager viewPager = view.findViewById(R.id.ls_view_pager);
> 			viewPager.setAdapter(new ByuFragmentPagerAdapter(getFragmentManager(), new ByuFragment[]{CourseListFragment.newInstance(studentCourses, "My Classes", currentYearTerm),
> ```

If this is getting long, put each fragment on it's own line. I this I like that anyways, so that people can see what all this is holding:
```java
viewPager.setAdapter........ new ByuFragment[] {
  frag1,
  frag2,
}));
```


---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 132 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30ipEukwxMhB5I4ak0:-L30ipEvBv8n0-UM1yOh:bn8mebd) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L132)):*
> ```Java
> 			ViewPager viewPager = view.findViewById(R.id.ls_view_pager);
> 			viewPager.setAdapter(new ByuFragmentPagerAdapter(getFragmentManager(), new ByuFragment[]{CourseListFragment.newInstance(studentCourses, "My Classes", currentYearTerm),
> 					CourseListFragment.newInstance(instructorCourses, "Courses I Teach", currentYearTerm)}));
> ```

If you're passing in the yearterm here, how do you manage it changing?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 35 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iyXYlMVU18lmJnJ7:-L30iyXYlMVU18lmJnJ8:b-48egti) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L35)):*
> ```Java
> 	public static final String CURRENT_YEAR_TERM_TAG = "yearTerm";
> 
> 	Map<YearTerm, ArrayList<Course>> coursesMap;
> ```

Should all of these be private?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 41 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30j0n6ZHNS_GfJkrCK:-L30j0n6ZHNS_GfJkrCL:bacw68w) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L41)):*
> ```Java
> 	private Spinner spinner;
> 
> 	public static CourseListFragment newInstance(HashMap<YearTerm, List<Course>> courseMap, String title, YearTerm currentYearTerm) {
> ```

Can this be a Map? I think it can...

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 41 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30k04QIwb3DNs1_zqw:-L30k04QIwb3DNs1_zqx:bi1li2n) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L41)):*
> ```Java
> 	private Spinner spinner;
> 
> 	public static CourseListFragment newInstance(HashMap<YearTerm, List<Course>> courseMap, String title, YearTerm currentYearTerm) {
> ```

Doesn't matter to me, but does it make things simpler to just have the Map being of <String, List<Course>>? It might be faster with the serialization...

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 57 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30jX0PlGAF9Z5A_Eyp:-L30jX0PlGAF9Z5A_Eyq:b-p4oljl) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L57)):*
> ```Java
> 		setHasOptionsMenu(true);
> 
> 		coursesMap = (HashMap<YearTerm, ArrayList<Course>>) getArguments().getSerializable(COURSES_TAG);
> ```

Map and List?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 81 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kIfIZrqrxBmPrVbT:-L30kIfIZrqrxBmPrVbU:bt65fh) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L81)):*
> ```Java
> 		SpinnerUtils.setUpAdapter(getActivity(), spinner, yearTerms, (position -> {
> 			selectedYearTerm = yearTerms.get(position);
> 			adapter.setList(coursesMap.get(selectedYearTerm));
> ```

Does setList not do the notifyDataSetChanged for you?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java, line 66 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kUYn0Muu5UagYzOK:-L30kUYn0Muu5UagYzOL:b-6nzv6e) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java#L66)):*
> ```Java
> 		@GET("coursebuilder/course/personEnrolled/{personId}/period/{yearTerm}")
> 		@Headers("Accept: application/json")
> 		Call<ArrayList<Course>> getCoursesForStudent(@Path("personId") String personId, @Path("yearTerm") String yearTerm);
> ```

Remove this?

---

*[app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java, line 161 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kuKSo71fmCTorTKA:-L30kuKSo71fmCTorTKB:bmkcvfs) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java#L161)):*
> ```Java
> 			// Sometimes the service gives us a different list of year terms. Here, we save the index
> 			// of the current yearTerm in order to properly display it in the yearTerm spinner
> 			if (currentYearTerm.equals(yearTermString)) {
> ```

I don't think I understand what this is doing and why we changed it... What is this about?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

geogor37 commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 37 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30fCZ62JsSvNagR-gb:-L34UB5Vw3v5Va61uIpm:b5516h3) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L37)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Is there a way that this could be saved in the spinner? How is the currentYearTerm
</blockquote></details>

Not here, that happens in the CourseListFragment.  I am changing it so that we just use the greatest value in the list we get back as the current yearterm, however.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 38 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30fSiGQsqE5SaJ5twN:-L34MR5kvAWCHjxb3FJG:buwyfky) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L38)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Use Map here instead of HashMap (always use the interface)
</blockquote></details>

We have to use a HashMap in order to pass it to the fragment

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 48 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30gFKsGQKKBttbmb0s:-L34bBN77sB85ag5TFGo:buqmnxz) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L48)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

This should be removed now. Just use the "greatest" year term returned from the service
</blockquote></details>

Done (sort of). I moved this into the block of code where we handle users entering the feature from My Next Class because we only need the current semester in those cases.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 76 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30gUTSDctFEIXhrugm:-L34bevqCQ1H6BQmBmp-:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L76)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Remove
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 83 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30h_5L_ZxhBXkg0JFO:-L34MqYlGmBcHTHo2JvE:b-4oev7t) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L83)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

use List
</blockquote></details>

It has to be an ArrayList in order to pass it to the activity

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 106 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30haZvN6aQeUAEOX_K:-L34MwZs5S0UByKiOQcy:b-ya81ng) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L106)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

use Map
</blockquote></details>

Same as above

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 107 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30hcGSc3ZURrkOt7Lr:-L34e2ESuUqz5XT_yFRZ:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L107)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Map and parameterize <YearTerm, List<Course>>
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 113 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iCkiV4ZTXyawGKaD:-L34e4ehFmOdbLG_McqM:blryzwc) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L113)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

If you parameterize, you shouldn't have to cast here
</blockquote></details>

It still makes me cast it for some reason; I'm not sure why, that seems weird to me too.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 131 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iQ9HX8cwgBaOj6Zt:-L34eFkJH_Nn9BUFjy1p:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L131)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

If this is getting long, put each fragment on it's own line. I this I like that anyways, so that people can see what all this is holding:
```java
viewPager.setAdapter........ new ByuFragment[] {
  frag1,
  frag2,
}));
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 132 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30ipEukwxMhB5I4ak0:-L34eI9o8Gdez1K0pMXw:blatsk8) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L132)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

If you're passing in the yearterm here, how do you manage it changing?
</blockquote></details>

No longer passing in YearTerm

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 35 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iyXYlMVU18lmJnJ7:-L34ePKBMHDyIOWOLU0I:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L35)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Should all of these be private?
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 41 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30j0n6ZHNS_GfJkrCK:-L34NDtwJKpAk6W2a2Ap:b-vj5ez1) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L41)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Can this be a Map? I think it can...
</blockquote></details>

Nope, same problem as above.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 41 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30k04QIwb3DNs1_zqw:-L34NMROxIDo_Au6HJeM:b-w0cfzy) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L41)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Doesn't matter to me, but does it make things simpler to just have the Map being of <String, List<Course>>? It might be faster with the serialization...
</blockquote></details>

I'm not sure; the reason I chose to pass it in as a YearTerm object is because then I can sort the list that gets passed into the spinner by the YearTerms (e.g. "20181" > "20175" > "20174" etc...)  rather than by the semester titles, although I guess I could wait to convert them to YearTerms until after retrieving the Map from the bundle of extras, is that what you're asking?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 57 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30jX0PlGAF9Z5A_Eyp:-L34ecy8Ynj3Vgka9LvQ:bnyxisy) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L57)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Map and List?
</blockquote></details>

It does need to be an ArrayList because the list gets passed to another activity as an argument later on, but I changed the HashMap to a Map.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java, line 81 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kIfIZrqrxBmPrVbT:-L34fH2v3mXl_amw_wOr:b997mi1) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/fragment/CourseListFragment.java#L81)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Does setList not do the notifyDataSetChanged for you?
</blockquote></details>

Haha whoops!  Yeah it does.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java, line 66 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kUYn0Muu5UagYzOK:-L34OgzNfIwz3ZDXBDE9:b-4vydow) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java#L66)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Remove this?
</blockquote></details>

We still use it when the user enters the feature from My Next Class

---

*[app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java, line 161 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kuKSo71fmCTorTKA:-L34OyDdsq74lVAq-bT3:b-qbiwaw) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java#L161)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

I don't think I understand what this is doing and why we changed it... What is this about?
</blockquote></details>

The changes I made were just to use the YearTerm constructor instead of the duplicate logic; I'm not sure why this bit of code was necessary, but it seems to me that the currentYearTerm may not always be the first one in the list given to the spinner, so this just returns the index of the currentYearTerm in order to display it properly in the spinner.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34u4MPBNMGGnoZelfR:-L34u4MPBNMGGnoZelfS:b-ckm053) (no related file):*
We also need to handle the case that someone has no classes. Do we use the ByuRecyclerViewActivity?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 113 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iCkiV4ZTXyawGKaD:-L34vBodpoA4Z_8f4A-Y:bycliwy) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L113)):*
<details><summary><i>Previously, geogor37 (Geoff Gorham) wrote…</i></summary><blockquote>

It still makes me cast it for some reason; I'm not sure why, that seems weird to me too.
</blockquote></details>

Talk to me. It's because you didn't parameterize correctly

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 60 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34tGCIkd7fiUup16rM:-L34tGCIkd7fiUup16rN:b-l7vv0u) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L60)):*
> ```Java
> 			enqueueCall(api.getCoursesForInstructor(personId), new CourseListCallback(this, list ->
> 					instructorCourses = createCourseMap(CollectionUtils.filter(list, course ->
> 							course == null || course.getSections().isEmpty()
> ```

We should have this null check on the student one as well (two lines above). Also, add a check for the sections being null before calling isEmpty

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 63 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34uJuC4ID63b3kTpIQ:-L34uJuC4ID63b3kTpIR:bydvzfu) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L63)):*
> ```Java
> 					))));
> 		} else {
> 			ControlDatesClientHelper.getCurrentYearTerm(this, new ByuCallback<ControlDatesResponseWrapper>(this) {
> ```

What is this? Why do we need this?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 104 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34ubKdzrlkVN34vwJs:-L34ubKdzrlkVN34vwJt:b58jdug) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L104)):*
> ```Java
> 		HashMap courseMap = new HashMap<YearTerm, List<Course>>();
> 		for (Course course : courses) {
> 			YearTerm courseYearTerm = new YearTerm(course.getPeriod());
> ```

Remove duplicate code with this:
```java
String yearTermCode = course.getPeriod() != null && course.getPeriod().length > 5 ? course.getPeriod().substring(0, 5);
YearTerm courseYearTerm = new YearTerm(yearTermCode);
```

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java, line 66 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kUYn0Muu5UagYzOK:-L34yKfr4CrQ-p2RRUTs:b-bmkr1r) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java#L66)):*
<details><summary><i>Previously, geogor37 (Geoff Gorham) wrote…</i></summary><blockquote>

We still use it when the user enters the feature from My Next Class
</blockquote></details>

I'm wanting to change that. Talk to me

---

*[app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java, line 161 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kuKSo71fmCTorTKA:-L34zDh71v5a34EonEjh:b-82pg7r) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java#L161)):*
<details><summary><i>Previously, geogor37 (Geoff Gorham) wrote…</i></summary><blockquote>

The changes I made were just to use the YearTerm constructor instead of the duplicate logic; I'm not sure why this bit of code was necessary, but it seems to me that the currentYearTerm may not always be the first one in the list given to the spinner, so this just returns the index of the currentYearTerm in order to display it properly in the spinner.
</blockquote></details>

Talk to me about this one

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

geogor37 commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34u4MPBNMGGnoZelfR:-L353x-cMUjTbHmZV3Pf:bsjaygc) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

We also need to handle the case that someone has no classes. Do we use the ByuRecyclerViewActivity?
</blockquote></details>

The Fragment extends LearningSuiteListFragment, which seems to have all of that built in

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 60 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34tGCIkd7fiUup16rM:-L354fT7skQZ3FJEnJTH:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L60)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

We should have this null check on the student one as well (two lines above). Also, add a check for the sections being null before calling isEmpty
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 63 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34uJuC4ID63b3kTpIQ:-L354hI3nrbIMpH70_tK:b-r0lpyy) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L63)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

What is this? Why do we need this?
</blockquote></details>

It's for when a user enters the feature from My Next Class, in which case we only need the classes from the current YearTerm

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 104 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34ubKdzrlkVN34vwJs:-L355J5K1WSNIPBeaUVl:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L104)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Remove duplicate code with this:
```java
String yearTermCode = course.getPeriod() != null && course.getPeriod().length > 5 ? course.getPeriod().substring(0, 5);
YearTerm courseYearTerm = new YearTerm(yearTermCode);
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34u4MPBNMGGnoZelfR:-L358R4tWIPhDC0fTZc0:b-13xr8j) (no related file):*
<details><summary><i>Previously, geogor37 (Geoff Gorham) wrote…</i></summary><blockquote>

The Fragment extends LearningSuiteListFragment, which seems to have all of that built in
</blockquote></details>

Please test this. Just clear your list of classes when they come back

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 63 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34uJuC4ID63b3kTpIQ:-L358bC-e41GLl6LfW39:b-oup360) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L63)):*
<details><summary><i>Previously, geogor37 (Geoff Gorham) wrote…</i></summary><blockquote>

It's for when a user enters the feature from My Next Class, in which case we only need the classes from the current YearTerm
</blockquote></details>

We can remove this. Talk to me

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

geogor37 commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34u4MPBNMGGnoZelfR:-L3AVWiOUw-LWB4vF06Z:b1bemxd) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Please test this. Just clear your list of classes when they come back
</blockquote></details>

Done.  We currently display "No published courses were found for the current semester" and then exit out of the feature.  I'm assuming we want to remove "for the current semester" now that we're going back further, right?

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 113 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30iCkiV4ZTXyawGKaD:-L3APCrOBtj2R88m36GQ:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L113)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Talk to me. It's because you didn't parameterize correctly
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java, line 63 at r2](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34uJuC4ID63b3kTpIQ:-L3APFj9jzNzWCsNKNR8:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/060224e408a280e519c3afed49b2d5c4c5166f88/app/src/main/java/edu/byu/suite/features/learningSuite/activity/LearningSuiteMainActivity.java#L63)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

We can remove this. Talk to me
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java, line 66 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kUYn0Muu5UagYzOK:-L3APESYPSiyx_F9OCFB:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java#L66)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

I'm wanting to change that. Talk to me
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java, line 161 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kuKSo71fmCTorTKA:-L35FoJiUWn8aJSb7YtI:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/mapMyClasses/MapMyClassesMainActivity.java#L161)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Talk to me about this one
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34u4MPBNMGGnoZelfR:-L3AXZ1u28YbADqt43cW:b-wzq709) (no related file):*
<details><summary><i>Previously, geogor37 (Geoff Gorham) wrote…</i></summary><blockquote>

Done.  We currently display "No published courses were found for the current semester" and then exit out of the feature.  I'm assuming we want to remove "for the current semester" now that we're going back further, right?
</blockquote></details>

Perfect :)

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java, line 66 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kUYn0Muu5UagYzOK:-L3AYCvE5KSTiV7ED81_:bf4n74c) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java#L66)):*
<details><summary><i>Previously, geogor37 (Geoff Gorham) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Can we remove this now?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

geogor37 commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L34u4MPBNMGGnoZelfR:-L3AYaG3IPXPFl9k1Dh7:b-896fix) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Perfect :)
</blockquote></details>

Done.

---

*[app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java, line 66 at r1](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501#-L30kUYn0Muu5UagYzOK:-L3AYl1InrMDsAHMP1cG:b-896fix) ([raw file](https://github.com/byu-oit/android-byu-suite-v2/blob/078babc50255edca7ac807e52fa4c730e05059b4/app/src/main/java/edu/byu/suite/features/learningSuite/service/LearningSuiteClient.java#L66)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Can we remove this now?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit/android-byu-suite-v2/501)*
<!-- Sent from Reviewable.io -->
